packages/kbn-eslint-plugin-eslint/rules/require_kibana_feature_privileges_naming.js

```
@@ -0,0 +1,227 @@
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+const ts = require('typescript');
+const path = require('path');
+
+function getImportedVariableValue(context, name, propertyName) {
+  try {
+    const parent = context
+      .getAncestors()
+      .find((ancestor) => ['BlockStatement', 'Program'].includes(ancestor.type));
+
+    if (!parent) return;
+
+    const importDeclaration = parent.body.find(
+      (statement) =>
+        statement.type === 'ImportDeclaration' &&
+        statement.specifiers.some((specifier) => specifier.local.name === name)
+    );
+
+    if (!importDeclaration) return;
+
+    const absoluteImportPath = require.resolve(importDeclaration.source.value, {
+      paths: [path.dirname(context.getFilename())],
+    });
+
+    const program = ts.createProgram([absoluteImportPath], {});
+    const sourceFile = program.getSourceFile(absoluteImportPath);
+
+    if (!sourceFile) return null;
+
+    const checker = program.getTypeChecker();
+    const symbols = checker.getExportsOfModule(sourceFile.symbol);
+    const symbol = symbols.find((s) => s.name === name);
+
+    if (!symbol) return null;
+
+    if (propertyName) {
+      const currentSymbol = checker.getTypeOfSymbolAtLocation(symbol, sourceFile);
+      const property = currentSymbol.getProperty(propertyName);
+
+      if (ts.isStringLiteral(property.valueDeclaration.initializer)) {
+        return property.valueDeclaration.initializer.text;
+      }
+
+      return null;
+    }
+
+    const initializer = symbol?.valueDeclaration?.initializer;
+
+    if (ts.isStringLiteral(initializer)) {
+      return initializer.text;
+    }
+
+    return null;
+  } catch (e) {
+    return null;
```

__question:__ Are there _expected_ and legitimate cases where the code in `try` might throw an exception? I’m wondering if swallowing errors like that could lead to the rule being silently broken and essentially ignored in the future?

---

makes sense, dropped try/catch in 80b91f3eabfbb71ae7bc9dc568fa0a516a9c9f66